src/components/ProductCard.tsx

```diff
@@ -3,13 +3,26 @@ import {
   Card,
   Button,
 } from 'react-bootstrap';
+import { connect, ConnectedProps } from 'react-redux'
+import { addToCart } from '../actions/cart';
 import { Product } from '../types/product';
 
-type Props = {
+const mapDispatch = {
+  onAddToCart: (product: Product) => addToCart(product),
+};
+
+const connector = connect(
+  null,
+  mapDispatch
+);
+
+type PropsFromRedux = ConnectedProps<typeof connector>;
+
+type Props = PropsFromRedux & {
   product: Product
 };
 
-const ProductCard = ({ product }: Props) => {  
+const ProductCard = ({ onAddToCart, product }: Props) => {  
```

Comment:
This can be typed as a React.FunctionalComponent, which provides stricter types to ensure that the props are passed through and typed correctly. (`FC` is a shorthand for `FunctionalComponent`)

```
const ProductCard: React.FC<Props> = () => {
```

Reply:
yep, I'll do this

`const ProductCard: FC<Props> = ({ onAddToCart, product }) => {  `

Reply:
Done.